### PR TITLE
Add WriteJSONWithStatus-function to write JSON with status

### PR DESCRIPTION
**Pull request by zwerch**
Self-explanatory function to add support for a different status code than 200, but with an own interface.

### CODE REVIEW

**Comment by bahlo** [comment]
Thanks for the PR! Maybe a name like `WriteJSONStatus` would be more ideomatic. 

And it would be great if you'd write a test for it. So we don't loose coverage :+1:

File: json.go [hunk 1]
```diff
@@ -25,3 +25,11 @@ func WriteJSON(w http.ResponseWriter, v interface{}) error {
 	w.Write(b)
 	return nil
 }
+
+// WriteJSONWithStatus writes the given statuscode into the header and the
+// given interface as JSON  or returns an error
+func WriteJSONWithStatus(w http.ResponseWriter, code int, v interface{}) {
+	w.WriteHeader(code)
+
+	WriteJSON(w, v)
```
Comment (bahlo) [inline json.go]: Can you please return the error from `WriteJSON` again, so it's possible to check for them?

Comment (zwerch) [inline json.go]: Oh, of course. Will do.

**Comment by bahlo** [comment]
Fixed in [`1.0.7`](https://github.com/bahlo/goat/commit/4de72452b8cfe7dc1dd9acabf3ca8d4d443dd1ad).

**Pull request closed without merging**